_posts/pl/2023-07-24-czy-wiesz-czym-jest-indexeddb.md

```diff
@@ -0,0 +1,187 @@
+---
+layout: post
+title: "Czy wiesz czym jest IndexedDB?"
+date: 2023-07-24T11:00:00+01:00
+published: true
+didyouknow: true
+lang: pl
+author: wstolarski
+image: /assets/img/posts/2023-07-24-czy-wiesz-czym-jest-indexeddb/post_img.jpg
+description:
+tags:
+  - indexeddb
+  - webstorage
+  - database
+  - javascript
+---
+
+IndexedDB to wbudowana w przeglądarkę internetową baza danych typu NoSQL. Przechowuje ona dane lokalnie w przeglądarce,
+co pozwala na korzystanie z nich nawet wtedy, gdy urządzenie nie ma połączenia z internetem. Dzięki temu IndexedDB
+stanowi świetną opcję dla aplikacji internetowych i jest dobrą alternatywą dla localStorage.
+
+W przeciwieństwie do localStorage, dane w IndexedDB nie są automatycznie usuwane przez przeglądarkę. Zostają one
+przechowywane trwale do momentu, gdy użytkownik zdecyduje się usunąć je ręcznie lub gdy aplikacja, która korzysta z
+IndexedDB, wykona odpowiednie operacje usuwania.
+
+Dodatkowo IndexedDB pozwala na:
+
+- obsługę transakcji,
+- tworzenie zapytań range queries,
+- obsługę indeksów,
+- przechowywanie znacznie większej ilości danych niż localStorage i cookies.
+
+## Kompatybilność i wsparcie przeglądarek
+
+Listę przeglądarek wspierających opisywany mechanizm możemy znaleźć [tutaj](https://caniuse.com/indexeddb).
+
+## Tworzenie bazy
+
+W celu rozpoczęcia prac z IndexedDB musimy najpierw "otworzyć" bazę danych (połączyć się z nią).
+
+```javascript
+//Otwieranie połączenia z bazą danych. Jeśli baza nie istnieje to zostanie utworzona
+
+const dbName = 'MyDatabase';
+const dbVersion = 1;
+const request = indexedDB.open(dbName, dbVersion);
+```
+
+Aby podejrzeć otworzoną bazę danych, musimy otworzyć konsolę deweloperską (devTools), przejść do zakładki "Application", następnie w sekcji "Storage" znajduje się "IndexedDB", a w niej utworzone bazy danych.
```

Comment:
Może wystarczy "Aby podejrzeć bazę danych" bez "otworzoną". To otworzoną i tak jest niezręczną kalką to nie powielajmy tego za dużo? ;)